test/specialfunctions/test_specialfunctions_gamma.fypp

```diff
@@ -1,7 +1,7 @@
-#:set WITH_QP = False
-#:set WITH_XDP = False
 #:include "common.fypp"
-#:set CI_KINDS_TYPES = INT_KINDS_TYPES + CMPLX_KINDS_TYPES
+#:set R_KINDS_TYPES = [KT for KT in REAL_KINDS_TYPES if KT[0] in ["sp","dp"]]
```

Comment:
@jvdp1 I didn't have time last week to look at that. I guess this should be done by propagating the fypp macros backwards in the dependency hierarchy but right now I do not see clearly how to do that without touching the build of test-drive itself or transferring the python preprocessing script to fpm.

Reply:
I guess it should be also in the CMake files. I will have a look.

Reply:
Should we then have one of the cmake builds test xdp and qp instead of the fpm build?

Reply:
Yes, good idea!
But the issue might still be there if someone try the tests with `fpm`, right? Or do I miss something?

Reply:
Yes, it would still be there. The ideas on top of my head are: add the python preprocessing to test-drive as well or add it to fpm (it might be the most robust solution but I'm guessing not the easiest)

Reply:
Passing the corresponding flags in the build `fpm test --profile release --flag '-DWITH_XDP -DWITH_QP'` are indeed enough to backpropagate the option.